topics/statistics/tutorials/RNN/tutorial.md

```diff
@@ -0,0 +1,364 @@
+to the vast availability of training data and enormous increases in computing capacity (Successfully 
+training complex neural networks in some domains requires lots of data and compute capacity). There 
+are various types of neural networks (Feedforward, recurrent, etc). In this tutorial, we discuss 
+the recurrent neural networks and explain how they differ from feedforward variants. We also describe 
+various RNN architectures and solve a sentiment analysis problem using RNN in Galaxy.       
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# Review of feedforward neural networks (FNN)
+
+In feedforward neural networks (FNN) all the fields of a training example are presented
+to the network at once, after which the the network generates an output. For example, a
+lung X-ray image is passed to a FNN, and the network predicts tumor or no tumor. By contrast,
+in RNN the training example fields are presented to the network one at a time. For example,
+a sequence of English words is passed to an RNN, one at a time, and the network generates a
+sequence of Persian words, one at a time. RNN handle sequential data, whether its temporal or ordinal.
+
+## Single layer FNN
+
+![Alternative text](../../images/FFNN_no_hidden.png "Single layer feedforward neural network")
+
+Figure 1 shows a single layer FNN, where the input is 3 dimensional. Each input field is multiplied by a
+weight. Afterwards, the results are summed up, along with a bias, and passed to an activation function.
+
+![Alternative text](../../images/activation.gif "Activation of the output neuron o1. Activation function f could be Sigmoid, Tanh, ReLU, etc.")
+
+The activation function can have many forms (sigmoid, tanh, ReLU, linear, step function, sign function, etc.).
+Output layer neurons usually have sigmoid or tanh functions.
+
+![Alternative text](../../images/sigmoid.gif "Sigmoid activation function")
+
+## Multi-layer FFN
+
+Minsky and Papert showed that a single layer FNN cannot solve problems in which the data is not linearly separable, 
+such as the XOR problem ({% cite Newell780 %}). Adding one (or more) hidden layers to FNN enables it to solve problems 
+in which data is non-linearly separable. Per Universal Approximation Theorem, a FNN with one hidden layer can represent 
+any function ({% cite Cybenko1989 %}), although in practice training such a model is very difficult (if not impossible), 
+hence, we usually add multiple hidden layers to solve complex problems.
+
+![Alternative text](../../images/FFNN.png "Feedforward neural network with a hidden layer. Biases to hidden/output layer neurons are omitted for clarity")
+
+## Learning algorithm 
+
+The learning algorithm incrementally tweaks the network weights, so that the network error on the training 
+set is minimized. Training set is composed of many training examples. Each training set is fed to the network 
```

Comment:
Please explain what a training set is. Why it is used this way and include the significance of test set as well.